src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -147,18 +150,28 @@ export class AccountManagementService implements IAccountManagementService {
 					if (accountResult.canceled === true) {
 						return;
 					} else {
+						this._telemetryService.createErrorEvent(TelemetryView.LinkedAccounts, TelemetryError.AddAzureAccountError, accountResult.errorCode,
+							this.getErrorType(accountResult.errorMessage))
+							.withAdditionalProperties({
+								'AuthLibrary': this._authLibrary
+							})
+							.send();
 						if (accountResult.errorCode && accountResult.errorMessage) {
 							throw new Error(localize('addAccountFailedCodeMessage', `{0} \nError Message: {1}`, accountResult.errorCode, accountResult.errorMessage));
-						} else if (accountResult.errorMessage) {
-							throw new Error(localize('addAccountFailedMessage', `{0}`, accountResult.errorMessage));
 						} else {
-							throw new Error(genericAccountErrorMessage);
+							throw new Error(accountResult.errorMessage ?? genericAccountErrorMessage);
 						}
 					}
 				}
 				let result = await this._accountStore.addOrUpdate(accountResult);
 				if (!result) {
 					this._logService.error('adding account failed');
+					this._telemetryService.createErrorEvent(TelemetryView.LinkedAccounts, TelemetryError.AddAzureAccountError, '-1',
```

Comment:
Generally we shouldn't be using the same view + error name in multiple places. This makes it harder to know what part exactly is failing when looking through the data.